sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetFilters.scala

```diff
@@ -42,6 +42,14 @@ private[parquet] class ParquetFilters(pushDownDate: Boolean, pushDownStartWith:
   private val makeEq: PartialFunction[DataType, (String, Any) => FilterPredicate] = {
     case BooleanType =>
       (n: String, v: Any) => FilterApi.eq(booleanColumn(n), v.asInstanceOf[java.lang.Boolean])
+    case ByteType =>
+      (n: String, v: Any) => FilterApi.eq(
+        intColumn(n),
+        Option(v).map(b => b.asInstanceOf[java.lang.Byte].toInt.asInstanceOf[Integer]).orNull)
```

Comment:
I think we can use `Byte` instead of `java.lang.Byte` here.

Reply:
I agree.

Also, there's no need to use `Option.map` because the value cannot be null. That's why the `IntegerType` case just casts the value.

Reply:
```scala
scala> null.asInstanceOf[Short].toInt.asInstanceOf[Integer]
res49: Integer = 0

scala> null.asInstanceOf[java.lang.Short].toInt.asInstanceOf[Integer]
java.lang.NullPointerException
  at scala.Predef$.Short2short(Predef.scala:360)
  ... 51 elided
```
That's why I use `Option.map` here.